common/djangoapps/third_party_auth/tasks.py

```diff
@@ -75,10 +78,31 @@ def fetch_saml_metadata():
                     num_changed += 1
                 else:
                     log.info(u"→ Updated existing SAMLProviderData. Nothing has changed.")
-        except Exception as err:  # pylint: disable=broad-except
-            log.exception(err.message)
-            num_failed += 1
-    return (num_changed, num_failed, len(url_map))
+        except (exceptions.SSLError, exceptions.HTTPError, exceptions.RequestException, MetadataParseError) as error:
```

Comment:
We should add an informative code comment here describing the use cases for the two exception handlers and the types of scenarios we expect to cover for each.